src/parser.rs

```diff
@@ -801,25 +806,12 @@ fn parse_jacoco_report_package<T: BufRead>(
         }
     }
 
-    for (class, result) in &results_map {
-        if result.lines.is_empty() && result.branches.is_empty() {
-            return Err(ParserError::InvalidData(format!(
-                "Class {}/{} is not the top class in its file.",
-                package, class
-            )));
-        }
-    }
-
-    // Change all keys from the class name to the file name and turn the result into a Vec.
-    // If package is the empty string, we have to trim the leading '/' in order to obtain a
-    // relative path.
+    // Change all the keys to use relative paths.
     Ok(results_map
         .into_iter()
         .map(|(class, result)| {
             (
-                format!("{}/{}.java", package, class)
-                    .trim_start_matches('/')
-                    .to_string(),
+                format!("{}/{}", package, class.trim_start_matches('/')).to_string(),
```

Comment:
I think you can fix the test failure by keeping the old behavior of trimming after format:
```suggestion
                format!("{}/{}", package, class).trim_start_matches('/').to_string(),
```

Reply:
I generally went much more file oriented than package/class oriented. I've since built more (and I'm overdue to open about 10 more PRs) where for example:
- Support go's weird format
- jacoco support for kotlin, scala, and java, including handling of things like missing function / method coverage
- Support for scoverage.xml
- Support for pycov's xml format

A few others. I need to do some work to pull my changes into something manageable as distinct PRs.

Reply:
@jasongilanfarr oh wow, thanks for your work!